homeassistant/components/freebox/router.py

```diff
@@ -0,0 +1,228 @@
+        """Initialize a Freebox router."""
+        self.hass = hass
+        self._entry = entry
+        self._host = entry.data[CONF_HOST]
+        self._port = entry.data[CONF_PORT]
+
+        self._api: Freepybox = None
+        self._name = None
+        self._mac = None
+        self._sw_v = None
+        self._attrs = {}
+
+        # Devices
+        self._devices: Dict[str, any] = {}
+
+        # Sensors
+        self._temperature_sensors: Dict[str, any] = {}
+        self._connection_sensors: Dict[str, any] = {}
+
+        self.listeners = []
+
+    async def setup(self) -> None:
+        """Set up a Freebox router."""
+        self._api = await get_api(self.hass, self._host)
+
+        try:
+            await self._api.open(self._host, self._port)
+        except HttpRequestError:
+            _LOGGER.exception("Failed to connect to Freebox")
+            return ConfigEntryNotReady
+
+        # System
+        fbx_config = await self._api.system.get_config()
+        self._mac = fbx_config["mac"]
+        self._name = fbx_config["model_info"]["pretty_name"]
+        self._sw_v = fbx_config["firmware_version"]
+
+        # Devices & sensors
+        await self.update_all()
+        async_track_time_interval(self.hass, self.update_all, SCAN_INTERVAL)
+
+    async def update_all(self, now: Optional[datetime] = None) -> None:
+        """Update all Freebox platforms."""
+        await self.update_sensors()
+        await self.update_devices()
+
+    async def update_devices(self) -> None:
+        """Update Freebox devices."""
+        new_device = False
+        fbx_devices: Dict[str, any] = await self._api.lan.get_hosts_list()
+
+        # Adds the Freebox itself
+        fbx_devices.append(
+            {
+                "primary_name": self.name,
+                "l2ident": {"id": self.mac},
+                "vendor_name": self.manufacturer,
+                "host_type": "router",
+                "active": True,
+                "attrs": self._attrs,
+            }
+        )
+
+        for fbx_device in fbx_devices:
+            device_mac = fbx_device["l2ident"]["id"]
+
+            if self._devices.get(device_mac) is None:
+                new_device = True
+
+            self._devices[device_mac] = fbx_device
+
+        async_dispatcher_send(self.hass, self.signal_device_update)
+
+        if new_device:
+            async_dispatcher_send(self.hass, self.signal_device_new)
+
+    async def update_sensors(self) -> None:
+        """Update Freebox sensors."""
+        # System sensors
+        syst_datas: Dict[str, any] = await self._api.system.get_config()
+        temperature_datas = {item["id"]: item for item in syst_datas["sensors"]}
+        # According to the doc it is only temperature sensors in celsius degree.
+        # Name and id of the sensors may vary under Freebox devices.
+
+        for sensor_key, sensor_attrs in temperature_datas.items():
+            self._temperature_sensors[sensor_key] = sensor_attrs["value"]
```

Comment:
Seems we can consolidate this iteration and the above iteration into a single iteration.
```suggestion
        for sensor in syst_datas["sensors"]:
            self._temperature_sensors[sensor["id"]] = sensor["value"]
```